funcx_sdk/tests/integration/test_serialization.py

```diff
@@ -0,0 +1,189 @@
+
+    return wrapper_double_first_argument
+
+
+@double_first_argument
+def decorated_add(a, b, c):
+    """
+    Third argument is ignored, only for decorator usage
+    """
+    return a + b
+
+
+def check_deserialized_foo(f):
+    p1, p2 = inspect.signature(f).parameters.items()
+
+    param_1: inspect.Parameter = p1[1]
+    param_2: inspect.Parameter = p2[1]
+
+    assert isinstance(param_1, inspect.Parameter)
+    assert isinstance(param_2, inspect.Parameter)
+    assert "x" == param_1.name
+    assert "y" == param_2.name
+    assert 3 == param_2.default
+
+
+def check_deserialized_bar(f):
+    p1, p2 = inspect.signature(f).parameters.items()
+
+    param_1: inspect.Parameter = p1[1]
+    param_2: inspect.Parameter = p2[1]
+
+    assert isinstance(param_1, inspect.Parameter)
+    assert isinstance(param_2, inspect.Parameter)
+    assert "a" == param_1.name
+    assert "b" == param_2.name
+    assert 5 == param_2.default
+
+
+def check_serialize_deserialize_foo(concrete_cls, pre_serialized=None):
+    serialized_foo = concrete_cls.serialize(foo)
+    deserialized_foo = concrete_cls.deserialize(serialized_foo)
+    check_deserialized_foo(deserialized_foo)
+
+    if pre_serialized:
+        deserialized_foo = concrete_cls.deserialize(pre_serialized)
+        check_deserialized_foo(deserialized_foo)
+
+
+def check_serialize_deserialize_bar(concrete_cls, pre_serialized=None):
+    def inner_bar(a, b=5):
+        return a * 5
+
+    serialized_bar = concrete_cls.serialize(inner_bar)
+    deserialized_bar = concrete_cls.deserialize(serialized_bar)
+    check_deserialized_bar(deserialized_bar)
+
+    if pre_serialized:
+        deserialized_bar = concrete_cls.deserialize(pre_serialized)
+        check_deserialized_bar(deserialized_bar)
+
+
+def check_serialize_deserialize_double(concrete_cls, pre_serialized=None):
+    serialized_add = concrete_cls.serialize(decorated_add)
+    deserialized_add = concrete_cls.deserialize(serialized_add)
+    assert 10 == deserialized_add(3, 4, True)
+    assert 7 == deserialized_add(3, 4, False)
+
+    if pre_serialized:
+        deserialized_add = concrete_cls.deserialize(pre_serialized)
+        assert 10 == deserialized_add(3, 4, True)
+        assert 7 == deserialized_add(3, 4, False)
+
+
+def test_base64_data():
+    jb = concretes.DillDataBase64()
+
+    d = jb.serialize(([2], {"y": 10}))
+    assert "00" == d.split("\n")[0]
+    args, kwargs = jb.deserialize(d)
+    assert args[0] == 2
+    assert kwargs["y"] == 10
+
+    # Depends on running this test in non 3.7 and passing
+    py37_serial_data = "00\ngANdcQBLAmF9cQFYAQAAAHlxAksKc4ZxAy4=\n"
+
+    args, kwargs = jb.deserialize(py37_serial_data)
+    assert args[0] == 2
+    assert kwargs["y"] == 10
+
+
+def test_pickle_deserialize():
+    jb = concretes.PickleCode()
+
+    try:
+        jb.serialize(foo)
+        raise AssertionError("Serialization using pickle should not be supported")
+    except NotImplementedError:
+        pass
```

Comment:
This looks like a use-case for `pytest.raises`, but it looks to me like this module has been written to run without pytest.

I think it's fine for things to only run under pytest, but I also don't think anything here *needs* to change.

Reply:
Yes, the module (and original test location) was written w/o pytest.  But the new location has another module which does use it, so no point to avoid it any longer, so I added pytest ref and simplified raise.